tests/api/cross_access/test_model.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Test the cross account request model."""
+from api.models import CrossAccountRequest, Employee
+from django.db import IntegrityError, transaction
+from django.test import TestCase
+from django.utils import timezone
+from tenant_schemas.utils import tenant_context
+
+from datetime import timedelta
+from unittest.mock import Mock
+from tests.identity_request import IdentityRequest
+
+
+class CrossAccountRequestModelTests(IdentityRequest):
+    """Test the cross account request model."""
+
+    def setUp(self):
+        """Set up the cross account request model tests."""
+        super().setUp()
+
+        self.ref_time = timezone.now()
+        self.employee = Employee.objects.create(
+            account="654321",
+            username="requestor_test",
+            first_name="requestor",
+            last_name="test",
+            email="requestor@redhat.com",
+        )
+        self.request = CrossAccountRequest.objects.create(
+            target_account="123456", employee=self.employee, endDate=self.ref_time + timedelta(10)
+        )
+
+    def tearDown(self):
+        """Tear down cross account request model tests."""
+        Employee.objects.all().delete()
+        CrossAccountRequest.objects.all().delete()
+
+    def test_request_creation_success(self):
+        """Test the creation of cross account request."""
+        self.assertEqual(self.request.target_account, "123456")
+        self.assertEqual(self.request.employee.account, "654321")
+        self.assertEqual(self.request.employee.username, "requestor_test")
+        self.assertIsNotNone(self.request.startDate)
+        self.assertEqual(self.request.endDate, self.ref_time + timedelta(10))
+        self.assertEqual(self.request.status, "requested")
+
+    def test_request_with_invalid_dates(self):
```

Comment:
@astrozzc @coderbydesign Just as an example, I created an example unit test that exercises the date validation logic a little more. Normally I've been including these types of tests in the iqe-rbac-plugin, but for simpler tests like these I think they are more appropriate as unit tests.

Reply:
Yeah that makes sense to me! 🎉

Reply:
Nice!!